tests/unit/Alignment.cpp

```diff
@@ -0,0 +1,355 @@
+    model::PrimitiveTypeKind::Number,
+    model::PrimitiveTypeKind::PointerOrNumber
+  };
+
+  for (model::ABI::Values ABIName : TestedABIs) {
+    const abi::Definition &ABI = abi::Definition::get(ABIName);
+    for (const auto &PrimitiveKind : RemainingTypes) {
+      for (std::uint64_t Size = 1; Size <= 16; Size *= 2) {
+        compareTypeAlignments(ABI,
+                              makePrimitive(Primitive::Generic, Size, *Binary),
+                              makePrimitive(PrimitiveKind, Size, *Binary));
+      }
+    }
+  }
+}
+
+static model::UnionField
+makeUnionField(std::uint64_t Index, model::QualifiedType Type) {
+  model::UnionField Result;
+  Result.Index() = Index;
+  Result.Type() = Type;
+  return Result;
+}
+
+// clang-format off
+template<typename ...FieldTypes>
+  requires (std::is_convertible_v<FieldTypes, model::UnionField> && ...)
+static model::QualifiedType makeUnion(model::Binary &Binary,
+                                      FieldTypes &&...Fields) {
+  // clang-format on
+  auto [Union, Path] = Binary.makeType<model::UnionType>();
+  (Union.Fields().emplace(std::forward<FieldTypes>(Fields)), ...);
+  return model::QualifiedType(Path, {});
+}
+
+BOOST_AUTO_TEST_CASE(UnionTypes) {
+  TupleTree<model::Binary> Binary;
+
+  using QT = model::QualifiedType;
+  QT Int16 = makePrimitive(model::PrimitiveTypeKind::Signed, 2, *Binary);
+  QT Int32 = makePrimitive(model::PrimitiveTypeKind::Signed, 4, *Binary);
+  QT Int64 = makePrimitive(model::PrimitiveTypeKind::Signed, 8, *Binary);
+  QT Float = makePrimitive(model::PrimitiveTypeKind::Float, 4, *Binary);
+  QT Double = makePrimitive(model::PrimitiveTypeKind::Float, 8, *Binary);
+  QT LongDouble = makePrimitive(model::PrimitiveTypeKind::Float, 16, *Binary);
+  QT WeirdLD = makePrimitive(model::PrimitiveTypeKind::Float, 12, *Binary);
+
+  for (model::ABI::Values ABIName : TestedABIs) {
+    const abi::Definition &ABI = abi::Definition::get(ABIName);
+
+    QT SimpleUnion = makeUnion(*Binary,
+                               makeUnionField(0, Int32),
+                               makeUnionField(1, Int64));
+    compareTypeAlignments(ABI, Int64, SimpleUnion);
+
+    QT SmallFloatUnion = makeUnion(*Binary,
+                                   makeUnionField(0, Int32),
+                                   makeUnionField(1, Float));
+    compareTypeAlignments(ABI, Int32, SmallFloatUnion);
+    compareTypeAlignments(ABI, Float, SmallFloatUnion);
+
+    QT BigFloatUnion = makeUnion(*Binary,
+                                 makeUnionField(0, LongDouble),
+                                 makeUnionField(1, Int64));
+    compareTypeAlignments(ABI, LongDouble, BigFloatUnion);
+
+    QT WeirdFloatUnion = makeUnion(*Binary,
+                                   makeUnionField(0, WeirdLD),
+                                   makeUnionField(1, Int32));
+    compareTypeAlignments(ABI, WeirdLD, WeirdFloatUnion);
+
+    QT EclipsedFloatUnion = makeUnion(*Binary,
```

Comment:
Eclipsed?

Reply:
I couldn't think of a better work: as in, the `float`'s layout does not matter because there's a more strictly-aligned field in there. Any ideas?

Reply:
`IrrelevantFloat`?

Leave a comment if you can't come up with a good name.

Reply:
I don't exactly like `irrelevant` :sadparrot:

At that point, maybe `latent` or `buried`?

Reply:
What about `supressed`?

Reply:
Works, but consider leaving a comment, no term is intuitive enough I think

Reply:
I went with `Eclipsed` plus a comment:
```cpp
    // Test the case where on top of the float field, there's also another
    // stricter-aligned field, which "eclipses" the tested field.
```

works?

Reply:
Yup